src/gui/gt_mdilauncher.cpp

```diff
@@ -438,19 +454,88 @@ GtMdiLauncher::open(const QString& id, GtObject* data, const QString& customId)
         icon = gt::gui::icon::layers();
     }
 
+    // custom widget to handle redocking
+    QPointer<GtDockableFrame> dockableWidget{qobject_cast<GtDockableFrame*>(wid)};
+    auto* tabWidget = qobject_cast<GtTabWidget*>(m_area.data());
+
+    // helper function to make mdi buttons
+    auto const makeMdiButton = [tabWidget, icon, dockableWidget,
+                                mdiItem = QPointer<GtMdiItem>(mdiItem)](){
+        assert(tabWidget);
+
+        // set custom close button
+        QPushButton* closeBtn = new QPushButton;
```

Comment:
```suggestion
        auto* closeBtn = new QPushButton;
```

In most parts of the code we use auto in this case